rich/console.py

```diff
@@ -723,6 +727,7 @@ def __init__(
         self._thread_locals = ConsoleThreadLocals(
             theme_stack=ThemeStack(themes.DEFAULT if theme is None else theme)
         )
+        self._buffer_indices_to_render: Set[int] = {self._thread_locals.buffer_index}
```

Comment:
Don't quite follow what the intent of this is. Handling nested captures? Can you explain to me on Tuesday please.

Reply:
Previously we only rendered content if the buffer index was 0. This was a hard-coded check. Now we also want to do this if we're inside a capture context (the buffer index is not 0) and where the `echo` arg is `True`, so we record the indices where that's the case.